src/main/java/org/firmata4j/fsm/EventPublisher.java

```diff
@@ -0,0 +1,38 @@
+package org.firmata4j.fsm;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author akia
+ * @since 2018-01
+ */
+public class EventPublisher {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventPublisher.class);
+    private static EventListener listener = null;
+
+    private EventPublisher() {
+    }
+
+    public static void setListener(EventListener listener) {
+        EventPublisher.listener = listener;
+    }
+
+    public static void publish(final Event event) {
+        if (listener == null) {
+            LOGGER.warn("received event without any listener being set");
+        } else {
+            new Thread(new Runnable() {
```

Comment:
Handilng every event in the new dedicated thread is a big overhead in my opinion. Events are frequent. This approasch can lead the client application to slow down because of garbage collection. The client application must decide on its own whether to do heavy handling in another thread or just do it quick.